src/main/java/com/aws/greengrass/clientdevices/auth/certificate/ServerCertificateGenerator.java

```diff
@@ -60,6 +60,11 @@ public synchronized void generateCertificate(Supplier<List<String>> connectivity
             return;
         }
 
+        if (certificateStore.getCaPrivateKey() == null) {
+            logger.atWarn().log("Not generating new server certificate as CA private key is not found");
+            return;
+        }
```

Comment:
This makes me super uncomfortable. We **_CANNOT_** silently fail like this. An error needs to be bubbled up so some action can be taken. Also consider increasing log level to error.
Can fix this as part of a separate PR.